File: src/main/java/com/dafttech/terra/engine/gui/anchors/AnchorCenterY.java
package com.dafttech.terra.engine.gui.anchors;

import com.dafttech.terra.engine.gui.GUIObject;
import com.dafttech.terra.engine.gui.containers.GUIContainer;

public class AnchorCenterY extends GUIAnchor {
    @Override
    public void applyAnchor(GUIObject object, GUIContainer container) {
        object.position.y = container.size.y / 2 - object.size.y / 2;
    }
}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/anchors/AnchorCenterY.scala
package com.dafttech.terra.engine.gui.anchors

import com.dafttech.terra.engine.gui.GUIObject
import com.dafttech.terra.engine.gui.containers.GUIContainer

class AnchorCenterY extends GUIAnchor {
  def applyAnchor(gObj: GUIObject, container: GUIContainer) {
    gObj.position = gObj.position.withY(container.size.y / 2 - gObj.size.y / 2)
  }
}